src/alr/alr-commands-test.adb

```diff
@@ -1,718 +1,155 @@
+         Trace.Warning
+           ("arguments cannot be forwarded to test runners when several "
+            & "exist.");
+         Trace.Warning ("use --id=<runner id> to call a specific test runner");
+      end if;
 
-         No_Args : constant Boolean := Args.Count = 0;
+      for Test_Setting of All_Settings loop
+         declare
+            use Alire.Directories;
+            use all type Ada.Strings.Unbounded.Unbounded_String;
 
-      begin
+            function Get_Args return AAA.Strings.Vector
+            is (if All_Settings.Length = 1 then Args
+                else AAA.Strings.Empty_Vector);
+            --  Only forward arguments if the runner is the only one.
 
-         --  We must go over all crates when listing is requested, or when we
-         --  need to match the search term against crate names. Otherwise, we
-         --  can directly retrieve the given crates.
+            S : constant Settings := Settings (Test_Setting);
 
-         if No_Args or else Cmd.Search then
-            for Crate of Alire.Index.All_Crates.all loop
-               if not Crate.Releases.Is_Empty then
-                  if No_Args or else Is_Match (Crate.Name) then
-                     if Cmd.Last then
-                        Candidates.Include (Crate.Releases.Last_Element);
-                     else
-                        for Release of Crate.Releases loop
-                           Candidates.Include (Release);
-                        end loop;
-                     end if;
-                  end if;
-               end if;
-            end loop;
-         else
-            for J in Args.First_Index .. Args.Last_Index loop
-               declare
-                  Allowed  : constant Alire.Dependencies.Dependency :=
-                               Alire.Dependencies.From_String (Args (J));
-                  Crate    : constant Alire.Crates.Crate :=
-                               Alire.Index.Crate (Allowed.Crate);
-                  Releases : constant Alire.Releases.Containers.Release_Set :=
-                               Crate.Releases;
-               begin
-                  for I in Releases.Iterate loop
-                     if Allowed.Versions.Contains (Releases (I).Version) then
-                        if not Cmd.Last or else
-                          I = Releases.Last or else
-                          not Allowed.Versions.Contains
-                            (Releases (Next (I)).Version)
-                        then
-                           Candidates.Include (Releases (I));
-                        end if;
-                     end if;
-                  end loop;
-               end;
-            end loop;
-         end if;
-      end Find_Candidates;
+            Dir      : constant Alire.Relative_Path := To_String (S.Directory);
+            Failures : Integer;
 
-   begin
-      --  Validate command line
+            Guard : Alire.Directories.Guard (Enter (Dir))
```

Comment:
Here, if the tests dir doesn't exist, it ends with  with
```
error: directory "tests" does not exist
error: alr encountered an unexpected error, re-run with -d for details.
error: error location: 0x5736818265a2 ada__directories__set_directory at ???
```
It would be nicer to just point out the missing dir and not consider this unexpected. Thinking of users that may edit the test directory.